plugins/auth-backend/src/lib/OAuthProvider.ts

```diff
@@ -224,6 +255,19 @@ export class OAuthProvider implements AuthProviderRouteHandlers {
     }
   }
 
+  identifyEnv(req: express.Request): string {
+    const reqEnv = req.query.env?.toString();
+    if (reqEnv) {
+      return reqEnv;
+    }
+    const stateParams = req.query.state?.toString();
+    if (!stateParams) {
+      return '';
+    }
+    const env = readState(stateParams).env;
+    return env ?? '';
```

Comment:
Best to return `undefined` here and above as well

Reply:
Updated.